internal/controller/controlplane_controller.go

```diff
@@ -176,29 +180,27 @@ func (r *ControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.ConfigMap{}).
 		Owns(&corev1.ServiceAccount{}).
 		Owns(&corev1.Secret{}).
-		Watches(&corev1.Secret{}, enqueueSecretsOfInterest()).
+		Watches(&corev1.Secret{}, enqueueSecretsOfInterest(mgr.GetLogger())).
 		Complete(r)
 }
 
-func enqueueSecretsOfInterest() handler.EventHandler {
+func enqueueSecretsOfInterest(logger logr.Logger) handler.EventHandler {
 	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
 		secret, ok := o.(*corev1.Secret)
 		if !ok {
 			return nil
 		}
 
 		// Check if the secret has the specified name
-		var matches []reconcile.Request
-		if secret.Name == util.VClusterKubeConfigSecret {
-			matches = append(matches, reconcile.Request{
-				NamespacedName: types.NamespacedName{
-					Name:      secret.GetName(),
-					Namespace: secret.GetNamespace(),
-				},
-			})
+		if secret.Name != util.VClusterKubeConfigSecret {
+			cpName, err := util.ControlPlaneNameFromNamespace(secret.Namespace)
+			if err == nil {
+				return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: cpName}}}
+			}
+			logger.Info("Ignoring non-ControlPlane Secret named "+util.VClusterKubeConfigSecret, "namespace", secret.Namespace)
```

Comment:
I may be missing something here, but it looks like there is no reconciliation if secret.Name == util.VClusterKubeConfigSecret - I thought that we wanted the controller to react when that secret was created, no ?

Reply:
Fixed.